src/System.Reflection/tests/FieldInfo/FieldInfo_MethodTests.cs

```diff
@@ -86,197 +39,123 @@ public void TestGetValue_Exception1()
         public void TestGetValue_Exception3()
         {
-            String fieldname = "nonstatic_strField";
-            FieldInfo fi = GetField(fieldname);
-            Object objInstance = new Object();
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-            Assert.Throws<System.ArgumentException>(() => { string retValue = (String)fi.GetValue(objInstance); });
-        }
-
-        //Verify SetValue method for static int FieldType
-        [Fact]
-        public void TestSetValue_StaticIntField()
-        {
-            String fieldname = "intFieldStatic";
-            int origfieldvalue = 0;
-            int newfieldValue = 1000;
-            FieldInfo fi = GetField(fieldname);
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-
-            origfieldvalue = (int)fi.GetValue(myInstance);
-            fi.SetValue(myInstance, (object)newfieldValue);
-            Assert.True(((int)fi.GetValue(myInstance)).Equals(newfieldValue));
-
-            //reset static field value to its original value
-            FieldInfoMethodTests.intFieldStatic = origfieldvalue;
-        }
-
-        //Verify SetValue method for non-static int FieldType
-        [Fact]
-        public void TestSetValue_IntField()
-        {
-            String fieldname = "intFieldNonStatic";
-            int origfieldvalue = 0;
-            int newfieldValue = 1000;
-            FieldInfo fi = GetField(fieldname);
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-
-            origfieldvalue = (int)fi.GetValue(myInstance);
-            fi.SetValue(myInstance, (object)newfieldValue);
-
-            Assert.True(((int)fi.GetValue(myInstance)).Equals(newfieldValue));
-        }
-
-        //Verify SetValue method for static String FieldType
-        [Fact]
-        public void TestSetValue_StaticStrField()
-        {
-            String fieldname = "static_strField";
-            String origfieldvalue;
-            String newfieldValue = "new";
-            FieldInfo fi = GetField(fieldname);
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-
-            origfieldvalue = (String)fi.GetValue(myInstance);
-            fi.SetValue(myInstance, (object)newfieldValue);
-
-            Assert.True(((String)fi.GetValue(myInstance)).Equals(newfieldValue));
-
-            //reset static field value to its original value
-            FieldInfoMethodTests.static_strField = origfieldvalue;
+            FieldInfo fi = GetAndTestFieldInfo("nonstatic_strField");
+            object objInstance = new object();
+            Assert.Throws<ArgumentException>(() => { string retValue = (string)fi.GetValue(objInstance); });
```

Comment:
Yeah, while others may disagree with me, I think verifying null is a good idea, if for no other reason than someone reading the code understands that you went through the trouble of determining the correct value here.  It also helps us to ensure that if in the future we do add a parameter name to the exception, the test will fail, and we can then update the test with the right name... without this null, the test wouldn't fail, and we'd likely neglect to add the verification.